pkg/storage/etcd/etcd_watcher.go

```diff
@@ -353,7 +354,7 @@ func (w *etcdWatcher) sendResult(res *etcd.Response) {
 		w.sendAdd(res)
 	case EtcdSet, EtcdCAS:
 		w.sendModify(res)
-	case EtcdDelete:
+	case EtcdDelete, EtcdExpire:
```

Comment:
Yes.  Before this change kube ignored the expire events from etcd.  I don't know there was anything that used etcd's ttl so it probably wasn't needed before.  An expire event is only when a key expires in etcd, so from kube's perspective it really should be handled like a delete.  The only difference between a delete and an expire event in etcd is who initiated the key removal.  For a delete, it is expressly asked by an external entity vs expire is done by etcd w/o external prompting